tests/js/app-tests.js

```diff
@@ -53,14 +53,40 @@ module.exports = {
         TestCase.assertInstanceOf(app, Realm.App);
     },
 
+    testNewAppFromString() {
+        let app = new Realm.App(config.id);
+        TestCase.assertInstanceOf(app, Realm.App);
+        TestCase.assertEqual(app.id, config.id);
+    },
+
+    testNewAppFromUndefined() {
+        const error = TestCase.assertThrows(() => {
```

Comment:
Why use the statement form `() => { stmt; }` rather than expression form `() => expr` here?